edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -342,6 +358,11 @@ async Task<Option<ServiceIdentity>> GetServiceIdentityInternal(string id)
 
         async Task HandleNoServiceIdentity(string id)
         {
+            if (this.lifeCycleSignal.IsCancellationRequested)
+            {
+                return;
+            }
```

Comment:
seems like this should be consistent with HandleNewServiceIdentity, either we check cancellation in both, or neither.

Reply:
Good catch. Missed `HandleNewServiceIdentity`